Moves can_synth to chemical_flags

spessbro opened the pull request:
Changes the can_synth var to be a bitflag in chemical_flags remember to
check for initial()

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Refactors the can_synth var in reagents to chemical_flags and changes the default to false so we don't get drug and reagent and other weird chems like that
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
oranges said good :)
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
refactor: refactored can_synth into chemical_flags
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

SgtHunk committed:
Fixes the lavaland syndie base air injector (#56835)

Fixes the air injector by making it part of an actual area in the syndie base, adds a plating on the tile the injector is on.
Creates a plating subtype with the lavaland atmos.
Replaces all var-edited platings in the lavaland syndicate base with the mentioned subtype.

tgstation-server committed:
Automatic changelog generation for PR #56835 [ci skip]

spessbro committed:
Moves can_synth to chemical_flags

Changes the can_synth var to be a bitflag in chemical_flags remember to
check for initial()

Thalpy commented on code/modules/reagents/chemistry/reagents/impure_reagents.dm:
```diff
@@ -23,6 +23,7 @@
 	name = "Toxic sludge"
 	description = "Toxic chemical isomers made from impure reactions. Causes toxin damage"
 	ph = 2
+	chemical_flags = REAGENT_CANSYNTH
```
Will this override the parent's flags?
i.e. all children of impurity should have `chemical_flags = REAGENT_INVISIBLE | REAGENT_SNEAKYNAME | REAGENT_DONOTSPLIT` as well

spessbro replied:
my brain smal

Thalpy replied:
Your brain strong!

spessbro committed:
unga

necromanceranne commented:
Since there is an absurd amount of reagents, wouldn't it make more sense to make the flag for what chems _can't_ be synthed?

XDTM commented:
considering the ones that shouldn't be synthed are the parents of all the others, i imagine it would be tricky unless they're handled by a separate blacklist

Jared-Fogle commented:
@necromanceranne 

> Since there is an absurd amount of reagents, wouldn't it make more sense to make the flag for what chems can't be synthed?

No, one of the motivations behind this is it makes it explicit that you want a chemical to be able to be synthesized. It is now impossible to do incidentally.

necromanceranne commented:
> considering the ones that shouldn't be synthed are the parents of all the others, i imagine it would be tricky unless they're handled by a separate blacklist

Oh right I forgot how our reagents inheritance is like. Fair enough.

LemonInTheDark committed:
Makes n2o nontoxic, for reasons (#56839)

Counter to the original intent of the change seen here #50126, n2o will currently deal damage when used with both internal tanks and floods, as the threshold picked for it was about 0.15 moles at room temperature. Even outside this oversight, the original goal can't easily be achieved. Because breathing works off partial pressures, anything you can do with an anesthetic can you can do with an n2o flood.

Therefore I don't think it's behavior worth keeping, as even as a way to disincentivize non-antag n2o floods it would do little.

tgstation-server committed:
Automatic changelog generation for PR #56839 [ci skip]

Rohesie commented on code/_globalvars/bitfields.dm:
```diff
@@ -301,6 +301,7 @@ DEFINE_BITFIELD(chemical_flags, list(
 	"REAGENT_INVISIBLE" = REAGENT_INVISIBLE,
 	"REAGENT_SNEAKYNAME" = REAGENT_SNEAKYNAME,
 	"REAGENT_SPLITRETAINVOL" = REAGENT_SPLITRETAINVOL,
+	"REAGENT_CANSYNTH" = REAGENT_CANSYNTH,
```
```suggestion
	"REAGENT_CAN_BE_SYNTHESIZED" = REAGENT_CAN_BE_SYNTHESIZED,
```
Clear names, please.

Rohesie reviewed (changes requested):
Just a small mass-replacement.

tf-4 committed:
Winter coats (#56854)

add: Thirteen more winter coats covering all departments and most jobs.

tgstation-server committed:
Automatic changelog generation for PR #56854 [ci skip]

spessbro committed:
repairing tgstation.dme

spessbro committed:
funny rename

spessbro committed:
yeah

spessbro committed:
ebin

spessbro commented:
no

Thalpy commented:
What

Rohesie commented:
> 
> 
> What

Their git-fu is weak. They had to remake the PR:
https://github.com/tgstation/tgstation/pull/56891

spessbro commented:
> Their git-fu is weak. They had to remake the PR:

how dare you out me like this
